lib/source/s3.js

```diff
@@ -0,0 +1,284 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const Aws = require('aws-sdk');
+
+const DEFAULT_INTERVAL = 60000;
+
+class S3Parser {
+  constructor() {
+    this.properties = {};
+  }
+
+  update(data) {
+    let properties = {};
+
+    try {
+      properties = JSON.parse(data.toString());
+    } catch (e) {
+      // We should have an error condition but tbd
+    }
+    this.properties = properties;
+  }
+}
+
+class S3 extends EventEmitter {
+  constructor(opts) {
+    super();
+    const options = opts || {};
+
+    if (!options.hasOwnProperty('bucket') || !options.hasOwnProperty('path')) {
+      throw new Error('Bucket or path not supplied');
+    }
+
+    if (!options.hasOwnProperty('name')) {
+      this.name = `s3-${options.bucket}-${options.path}`;
+    }
+
+    this.interval = DEFAULT_INTERVAL;
+    this._parser = options.parser || new S3Parser(options);
+    this._ok = false;
+    this._updated = null;
+
+    /**
+     * Initialize the s3 client
+     */
+    this.service = new Aws.S3({
+      region: Config.get('index:region')
+    });
+
+    this.configure(options);
+    this.etag = null;
+    this.properties = {};
+  }
+
+  /**
+   * Check the status of the polling interval loop
+   *
+   * @return {Boolean} True if the pooling loop is running
+   */
+  get _running() {
+    return !!this._timer;
+  }
+
+  /**
+   * The source type
+   *
+   * @returns {string}
+   * @private
+   */
+  get _type() {
+    return this.constructor.type;
+  }
+
+  /**
+   * @param {Object} params
+   * @returns {Boolean}
+   */
+  configure(params) {
+    let changed = false;
+
+    changed = S3.setIfChanged(this, 'bucket', params.bucket || Config.get('index:bucket')) || changed;
+    changed = S3.setIfChanged(this, 'interval', params.interval || DEFAULT_INTERVAL) || false;
```

Comment:
Should this be `|| changed` at the end instead of `|| false` like the other two?